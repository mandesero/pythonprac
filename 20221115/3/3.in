from time import time

cur = time()
alp = Alpha(c=10, z=2, a=42, b='qw', d =[7217109,21791,1589,'218'])
alp.e = -214
for a in alp.d:
	print(a*4)
print(alp)

for c in [chr(i) for i in range(ord('a'), ord('z') + 1)]:
	size = 128000
	l = []
	for i in range(size):
		l.append(i)
	setattr(alp, c, l)

s = 0
for c in [chr(i) for i in range(ord('a'), ord('z') + 1)]:
	s += sum(getattr(alp, c))
print(s)

t1 = time() - cur

cur = time()
alq = AlphaQ(c=10, z=2, a=42, b='qw', d =[7217109,21791,1589,'218'])
alq.e = -214
for a in alq.d:
	print(a*4)
print(alq)


for c in [chr(i) for i in range(ord('a'), ord('z') + 1)]:
	size = 128000
	l = []
	for i in range(size):
		l.append(i)
	setattr(alq, c, l)

s = 0
for c in [chr(i) for i in range(ord('a'), ord('z') + 1)]:
	s += sum(getattr(alq, c))
print(s)
t2 = time() - cur
print(t2 > t1, t1, t2)
