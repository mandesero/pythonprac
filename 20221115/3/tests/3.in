from time import time
from string import ascii_lowercase

t0 = time()
a = Alpha()
for i in range(1000):
    for l in ascii_lowercase:
        setattr(a, l, i)
print(a)
t1 = time() - t0
t0 = time()

a = AlphaQ()
for i in range(1000):
    for l in ascii_lowercase:
        setattr(a, l, i)
print(a)
t2 = time() - t0

print(t1 < t2)