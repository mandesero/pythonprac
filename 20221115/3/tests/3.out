a: 999, b: 999, c: 999, d: 999, e: 999, f: 999, g: 999, h: 999, i: 999, j: 999, k: 999, l: 999, m: 999, n: 999, o: 999, p: 999, q: 999, r: 999, s: 999, t: 999, u: 999, v: 999, w: 999, x: 999, y: 999, z: 999
a: 999, b: 999, c: 999, d: 999, e: 999, f: 999, g: 999, h: 999, i: 999, j: 999, k: 999, l: 999, m: 999, n: 999, o: 999, p: 999, q: 999, r: 999, s: 999, t: 999, u: 999, v: 999, w: 999, x: 999, y: 999, z: 999
True
